src/sensor/pingsensor.cpp

```diff
@@ -22,8 +24,9 @@ void PingSensor::request(int id)
     }
     qCDebug(PING_PROTOCOL_PINGSENSOR) << "Requesting:" << id;
 
-    ping_message_empty m;
-    m.set_id(id);
+    ping_message m(10);
+    m.set_payload_length(0);
```

Comment:
Wait, why is this necessary ? Why is not zero by default ?

Reply:
msgData is allocated on the heap and not zero initialized, nor is it initialized by this message constructor, because the constructor doesn't know anything about the message payload or it's length.

Reply:
Our `ping_message_empty` constructor did the initialization for us manually in the past.